### ISSUE TITLE
Documentation: Organize filters into single-page categories by their general use case

### ISSUE THREAD

**Issue by ADTC** [Shopify/liquid#1409]
@dylanahsmith @pushrax thank you for approvals and merging #1408. I wish to contribute another PR for better organizing the filters in the official docs, where it is currently a long alphabetical list of all types of filters, and each filter has its own page. Compare this to Shopify's version:

1. Filters are categorized by their general use case.
2. All filters within a single category are placed on a single page.
3. There's an introduction on the page before the first filter.

![image](https://user-images.githubusercontent.com/6047296/110352901-6dfeb180-8071-11eb-9ff7-61e76f1bedf9.png)

If it is something you will accept, I'll work on reorganizing the documentation of filters to a similar format. 🙂

**Comment by dylanahsmith** [comment]
There are trade-offs to having one or the other.  Grouping related filters or tags by category helps with discovery when the reader doesn't know what is available.  On the other hand, the reader might want reference documentation for a tag or filter where they know the name, such as when they see it used in a template written by someone else.

Right now the documentation seems inconsistent by having tags grouped by category and filters not grouped.

Perhaps we could solve both use cases by having tags and filters grouped by category in the sidebar, while also having an index page that links to all tags and/or filters to use for reference documentation.

**Comment by ADTC** [comment]
Yes, I realized that with the current setup, each filter gets an URL that isn't an anchor link. I think we will be fine if we can redirect these page based URLs to anchor link based URLs, so that old references do not break. We can have the same setup for tags.

For example:
```
https://shopify.github.io/liquid/filters/abs/
```
redirects to:
```
https://shopify.github.io/liquid/filters/math/#abs
```
Is such redirection possible?

It's also necessary to have a summary listing of each subheading in the page. In Shopify, this appears on the right side. This might be a good setup to replicate.

**Also:** Tags section needs some restructuring as well. _Raw_ and _Comment_ pages should be removed, and replaced with a single _Template_ page, which documents all of these: 
```
comment
echo
include
liquid
raw
render
```
And the `ifchanged` tag needs to be documented under Control Flow or Iteration _(which one?)_. [This documentation looks good](https://help.marketpath.com/liquid/ifchanged) but I'm not sure if we'd have permission to copy.

Finally, aren't there any objects we can document? The `forloop` object comes to mind.

**Comment by dylanahsmith** [comment]
Good point about redirects.  It looks like we can use anchor redirects.  For instance, I tried changing the https://shopify.github.io/liquid/tags/ redirect to instead redirect to https://shopify.github.io/liquid/tags/control-flow/#unless locally as a proof of concept and it worked with this change

```diff
diff --git a/_tags/control-flow.md b/_tags/control-flow.md
index 83f30fc3..3a1a7651 100644
--- a/_tags/control-flow.md
+++ b/_tags/control-flow.md
@@ -1,7 +1,6 @@
 ---
 title: Control flow
 description: An overview of control flow and conditional tags in the Liquid template language.
-redirect_from: /tags/
 ---
 
 Control flow tags can change the information Liquid shows using programming logic.
diff --git a/pages/redirects/tags/index.html b/pages/redirects/tags/index.html
new file mode 100644
index 00000000..61b07276
--- /dev/null
+++ b/pages/redirects/tags/index.html
@@ -0,0 +1,4 @@
+---
+permalink: /tags
+redirect_to: /tags/control-flow/#unless
+---
```

> It's also necessary to have a summary listing of each subheading in the page. In Shopify, this appears on the right side. This might be a good setup to replicate.

:+1: Just pay attention to how it would look on mobile

> **Also:** Tags section needs some restructuring as well. _Raw_ and _Comment_ pages should be removed, and replaced with a single _Template_ page

Yeah, those single tag pages seemed out of place.

> And the `ifchanged` tag needs to be documented under Control Flow or Iteration _(which one?)_.

It doesn't do any iteration, so Control Flow seems the most suitable.

> [This documentation looks good](https://help.marketpath.com/liquid/ifchanged) but I'm not sure if we'd have permission to copy.

They would have to provide a license giving permission to distribute a copy of it.

> Finally, aren't there any objects we can document? The `forloop` object comes to mind.

It probably makes sense to document the `forloop` object along with the `for` tag.  Another one is `tablerowloop` for the `tablerow` tag.  These only make sense inside these tags, so it doesn't make sense to have independent object documentation.

**Comment by ADTC** [comment]
Great! Noted everything.

It looks like, the object should be `tablerow` [according to this](https://shopify.dev/docs/themes/liquid/reference/objects/tablerow) and not `tablerowloop`. Can I get a confirmation?

Most of the [objects listed here](https://shopify.dev/docs/themes/liquid/reference/objects) are Shopify-specific but I'm suspecting `request` and `template` could be standard Liquid objects. Is there a code block where this can be verified?

**Comment by ADTC** [comment]
I should note that there's a major documentation update ongoing in #1403 and we should avoid duplicate work and conflicts.

**Comment by microalps** [comment]
> And the ifchanged tag needs to be documented under Control Flow or Iteration (which one?). [This documentation looks good](https://help.marketpath.com/liquid/ifchanged) but I'm not sure if we'd have permission to copy.

Looks like this is still missing. I see #1403 is closed so it shouldn't block further work.
